base/float.jl

```diff
@@ -360,6 +360,7 @@ ceil(::Type{T}, x::AbstractFloat) where {T<:Integer} = trunc(T,round(x, RoundUp)
 ceil(::Type{T}, x::Float16) where {T<:Integer} = ceil(T, Float32(x))
 round(::Type{T}, x::AbstractFloat) where {T<:Integer} = trunc(T,round(x, RoundNearest))
 round(::Type{T}, x::Float16) where {T<:Integer} = round(T, Float32(x))
+round(::Type{T}, x::Real) where {T<:AbstractFloat} = T(round(x))
```

Comment:
Could this be done as `round(::Type{T}, x::Real) where {T<:Real} = T(round(x))` (that would avoid one special case for `Rational`)? Or does that cause problems elsewhere?

Reply:
I added it @simonbyrne :+1: